### PR TITLE
Remove some cloaking for stricter null checks and fix a doc block.

### PR DESCRIPTION
**Pull request by dereuromark**
Small fixes.

### CODE REVIEW

**Comment by markstory** [comment]
Teats are failing.

**Comment by dereuromark** [comment]
Looks like the last 3.next merge broak some tests, as well?

**Comment by markstory** [comment]
I think there has been a change in PHPUnit as well. I noticed some other pull requests have started hitting failures due to not being able to clone an object in FormHelper tests.

**Comment by markstory** [comment]
The current builds for the 3.next branch are passing. So I think some of the errors/failures are coming from your changes.

**Comment by dereuromark** [comment]
That's super weird, I specifically looked into object attributes, that only can/should be null if not existing.
The code functionally should behave the same, or some voodoo is going on here.
@thinkingmedia Do your eyes spot anything I am missing maybe?

File: src/View/View.php [hunk 1]
```diff
@@ -334,7 +334,7 @@ public function __construct(
         $this->eventManager($eventManager);
         $this->request = $request ?: Router::getRequest(true);
```
Comment (thinkingmedia) [inline src/View/View.php]: @dereuromark `Router::getRequest(true)` is returning `false`.

File: src/Database/Query.php [hunk 2]
```diff
@@ -1605,7 +1605,7 @@ public function newExpr($rawExpression = null)
      */
     public function func()
     {
-        if (empty($this->_functionsBuilder)) {
+        if ($this->_functionsBuilder === null) {
```
Comment (thinkingmedia) [inline src/Database/Query.php]: should be okay, but maybe assign `null` up above. Could use falsey but I find that to be a variant on the empty.

File: src/Event/EventDispatcherTrait.php [hunk 3]
```diff
@@ -48,7 +48,7 @@ public function eventManager(EventManager $eventManager = null)
     {
         if ($eventManager !== null) {
             $this->_eventManager = $eventManager;
-        } elseif (empty($this->_eventManager)) {
+        } elseif ($this->_eventManager === null) {
```
Comment (thinkingmedia) [inline src/Event/EventDispatcherTrait.php]: looks good to me.

File: src/TestSuite/IntegrationTestCase.php [hunk 4]
```diff
@@ -945,7 +945,7 @@ public function assertSession($expected, $path, $message = '')
      */
     public function assertCookie($expected, $name, $message = '')
     {
-        if (empty($this->_response)) {
+        if (!$this->_response) {
```
Comment (thinkingmedia) [inline src/TestSuite/IntegrationTestCase.php]: we look for wider coverage in tests. I would prefer the empty. This is a test after all and empty covers more edge cases.

Comment (dereuromark) [inline src/TestSuite/IntegrationTestCase.php]: See the other places in this class.

File: src/View/View.php [hunk 5]
```diff
@@ -334,7 +334,7 @@ public function __construct(
...
         $this->response = $response ?: new Response();
-        if (empty($this->request)) {
+        if ($this->request === null) {
```
Comment (dereuromark) [inline src/View/View.php]: Thanks!

**Comment by dereuromark** [comment]
Should be good now.

PS: In the future a class attribute for an object should always just have two cases
- null (empty)
- the object

never a third (bool) state IMO.
Especially since/when this is not documented anywhere.

**Comment by dereuromark** [comment]
Actually, end() returning false here and leaking into other objects trusting the return value declared was probably an oversight.
More strict checks now revealed that.

I removed the false return value for the doc bock to state the truth again, but we should keep the checks as they are.
